ts/lib220.ts

```diff
@@ -1,5 +1,79 @@
 import { getRunner } from './runtime';
 
+function hexColorChannel(n: number): string {
+  let v = (Math.floor(n * 255)).toString(16);
+  if (v.length < 2) v = '0' + v;
+  return v;
+}
+
+function rgbToHex(rgb: number[]): string {
+  let hex = '#';
+  for (let i = 0; i < 3; ++i) {
+    hex += hexColorChannel(rgb[i]);
+  }
+  return hex;
+}
+
+export class DrawingCanvas {
+  width: number = 1;
+  height: number = 1;
+  ctx: CanvasRenderingContext2D | undefined = undefined;
+  constructor(w: number, h: number) {
+    if (arguments.length !== 2) {
+      throw new TypeError(`Failed to construct Node 'DrawingCanvas': 2 arguments required but ${arguments.length} given`);
+    }
+    this.width = w;
+    this.height = h;
+    if (typeof document === 'undefined') {
+      return; //  for node
+    }
+    const canvases = document.getElementById('canvases')!;
```

Comment:
So, some weird effects will occur if a program chooses to show an image and use drawingCanvas at the same time. I think we should just be aware of it and totally ignore it for now. This exists only for in-class demos.

Reply:
The drawing actually happens in a saved context `ctx`, so creation of new canvases is not a problem - it still draws on the correct canvas (the one that was created in the constructor).

What effect were you thinking of? The one edge case that we have is that we delete older canvases. Interestingly, the browser happily accepts drawing commands to a context which no longer exists, without any errors.

Reply:
I understand it will draw to the right canvas, but I was thinking about interleaving.